overlord/ifacestate/ifacemgr.go

```diff
@@ -558,8 +569,32 @@ func (m *InterfaceManager) initUDevMonitor() error {
 	return nil
 }
 
-// initInterfacesRequestsManager should only be called if prompting is
-// supported and enabled.
+// interfacesRequestsControlHandlerServicePresent returns true if there is at
+// least one snap which has a "snap-interfaces-requests-control" connection
+// with an app declared by the "handler-service" attribute.
+//
+// The caller must ensure that the state lock is held.
+func (m *InterfaceManager) interfacesRequestsControlHandlerServicePresent() (bool, error) {
+	return interfacesRequestsControlHandlerServicePresentImpl(m)
+}
+
+var interfacesRequestsControlHandlerServicePresentImpl = func(m *InterfaceManager) (bool, error) {
```

Comment:
I want the function to be a method but I want to be able to mock the implementation as a function which can take a minimal `InterfaceManager` wrapper around a custom state. But in retrospect, I think I was trying to have it both ways: either the method itself should be mockable or the implementation should simply take a `state.State` instead of an `InterfaceManager`. Would you prefer the former?